client/network/src/service.rs

```diff
@@ -381,7 +384,19 @@ where
 				}
 			};
 
-			let mut builder = SwarmBuilder::new(transport, behaviour, local_peer_id)
+			let builder = if let Some(spawner) = params.executor {
+				struct SpawnImpl<F>(F);
+				impl<F: Fn(Pin<Box<dyn Future<Output = ()> + Send>>)> Executor for SpawnImpl<F> {
+					fn exec(&self, f: Pin<Box<dyn Future<Output = ()> + Send>>) {
+						(self.0)(f)
+					}
+				}
+				SwarmBuilder::with_executor(transport, behaviour, local_peer_id, SpawnImpl(spawner))
+			} else {
+				let tp = ThreadPoolBuilder::new().name_prefix("libp2p-swarm-task-").create()?;
+				SwarmBuilder::with_executor(transport, behaviour, local_peer_id, tp)
+			};
```

Comment:
Is this related to WebRTC? If not, I'd prefer if this was a separate PR.

Reply:
This is what we need to upgrade to libp2p 0.50. I believe it would be nice to create a separate PR so we can share the code.

Reply:
> Is this related to WebRTC? If not, I'd prefer if this was a separate PR.

https://github.com/paritytech/substrate/pull/12734

Reply:
But again, if I remove the above code, the PR will no longer compile 😕

Reply:
What if you set the base branch of this branch to point to the libp2p upgrade branch?